### ISSUE TITLE
[BUG]  Seems method (bg *BroadcastGroup) splitRequest  have a little typo 

### ISSUE THREAD

**Issue by tantra35** [go-graphite/carbonapi#570]
**Describe the bug**
When we decided to switch to use `maxBatchSize: 100` from  `maxBatchSize: 0` we have see abnormal request count from carbonapi to our go-carbonservers, this count incresed up to 100 times. Its abnormal due we doesn't use so huge wildcard request, as one might think

So we inspected sources a little bit and make a little change in `splitRequest` method(by our oppinion it have a bug - typo?)

```patch
 zipper/broadcast/broadcast_group.go | 27 +++++++++++++++++----------
 1 file changed, 17 insertions(+), 10 deletions(-)

diff --git a/zipper/broadcast/broadcast_group.go b/zipper/broadcast/broadcast_group.go
index 775b9377..cde3ab26 100644
--- a/zipper/broadcast/broadcast_group.go
+++ b/zipper/broadcast/broadcast_group.go
@@ -152,9 +152,10 @@ func (bg *BroadcastGroup) doMultiFetch(ctx context.Context, logger *zap.Logger,
 		return
 	}
 
+	logger = logger.With(zap.String("backend_name", backend.Name()), zap.Int("splited requests", len(requests)))
+
 	for _, req := range requests {
 		go func(req *protov3.MultiFetchRequest) {
-			logger = logger.With(zap.String("backend_name", backend.Name()))
 			logger.Debug("waiting for slot",
 				zap.Int("max_connections", bg.limiter.Capacity()),
 			)
@@ -204,7 +205,7 @@ func (bg *BroadcastGroup) doSingleFetch(ctx context.Context, logger *zap.Logger,
 		return
 	}
 
-	logger = logger.With(zap.String("backend_name", backend.Name()))
+	logger = logger.With(zap.String("backend_name", backend.Name()), zap.Int("splited requests", len(requests)))
 	logger.Debug("waiting for slot",
 		zap.Int("max_connections", bg.limiter.Capacity()),
 	)
@@ -241,9 +242,15 @@ func (bg *BroadcastGroup) splitRequest(ctx context.Context, request *protov3.Mul
 	}
 
 	var requests []*protov3.MultiFetchRequest
+	newRequest := &protov3.MultiFetchRequest{}
+
 	var err merry.Error
 	for _, metric := range request.Metrics {
-		newRequest := &protov3.MultiFetchRequest{}
+		if len(newRequest.Metrics) >= backend.MaxMetricsPerRequest() {
+			requests = append(requests, newRequest)
+			newRequest = &protov3.MultiFetchRequest{}
+		}
+
 		// TODO(Civil): Tags: improve logic
 		if strings.HasPrefix(metric.Name, "seriesByTag") {
 			newRequest.Metrics = append(newRequest.Metrics, protov3.FetchRequest{
@@ -253,20 +260,20 @@ func (bg *BroadcastGroup) splitRequest(ctx context.Context, request *protov3.Mul
 				PathExpression:  metric.PathExpression,
 				FilterFunctions: metric.FilterFunctions,
 			})
-			requests = append(requests, newRequest)
+
 			continue
 		}
 
 		// Do not send Find requests if we have neither globs in the request nor metric expansions
 		if !strings.ContainsAny(metric.Name, "*{") {
 			newRequest.Metrics = append(newRequest.Metrics, protov3.FetchRequest{
-				Name:            metric.PathExpression,
+				Name:            metric.Name,
 				StartTime:       metric.StartTime,
 				StopTime:        metric.StopTime,
 				PathExpression:  metric.PathExpression,
 				FilterFunctions: metric.FilterFunctions,
 			})
-			requests = append(requests, newRequest)
+
 			continue
 		}
 
@@ -318,16 +325,16 @@ func (bg *BroadcastGroup) splitRequest(ctx context.Context, request *protov3.Mul
 					FilterFunctions: metric.FilterFunctions,
 				})
 
-				if len(newRequest.Metrics) == backend.MaxMetricsPerRequest() {
+				if len(newRequest.Metrics) >= backend.MaxMetricsPerRequest() {
 					requests = append(requests, newRequest)
 					newRequest = &protov3.MultiFetchRequest{}
 				}
 			}
 		}
+	}
 
-		if len(newRequest.Metrics) > 0 {
-			requests = append(requests, newRequest)
-		}
+	if len(newRequest.Metrics) > 0 {
+		requests = append(requests, newRequest)
 	}
 
 	return requests, err
```

After that we see dramaticaly reduce recouest count to our go-carbon servers, and many widlcard request begin work as expected

This patch make 2 major things:

1. Make `splitRequest` reinterable on the same metric set. So if we pass already processed metrics(after expand wildcard throw find) we have got the same "batches", as after find

2. remove typo, by doing transformation `Name:            metric.PathExpression,` => `Name:            metric.Name,` on 
https://github.com/go-graphite/carbonapi/blob/main/zipper/broadcast/broadcast_group.go#L263

**CarbonAPI Version**
1.4.2.1

**Comment by Civil** [comment]
That indeed looks like a bug, but I think it would be better if you just make a Pull Request against carbonapi, to preserve who proposed/made the change in a changelog

**Comment by Civil** [comment]
As I've just merged your PR I think it's ok to close the issue.

P.S. To do that automatically you can add something like
`Fixes #570` in the text of a commit and github would automatically close corresponding issue.
